### PR TITLE
DEV-100: update readme

### PR DESCRIPTION
**Pull request by mpark63**
### In preparation for recruitment. 

added sections: 
- about 
- structure 
- instructions for running locally 
- note on testing 

very open to suggestions!

# view below: 
## uCredit-API
This is the backend for [uCredit](https://ucredit.me), 4-year course planning, streamlined. Deployed on [Render](render.com).

- Development deployment: https://ucredit-dev.onrender.com/
- Production deployment: https://ucredit-api.onrender.com/

Please allow a few seconds for the server to start up. 

## Structure
- API routes saved in ./routes
- Mongoose Schema saved in ./model
- Tests saved in ./tests
- Database connection made in ./data/db.js

### Run locally
1. Clone this repository. 
```
git clone https://github.com/uCredit-Dev/uCredit-API.git
``` 
2. Install dependencies. 
```
npm i
``` 
3. Create .env file at the root. Copy paste .env for development! (Ask a dev for access)
```
touch .env
``` 
4. Start server.
```
npm start
``` 
5. Navigate to http://localhost:4567/

### Run locally with docker 

To run program from a fresh clone:

1. Create a `.env` file with the following content:

```
DEBUG=True
SIS_API_KEY=YOUR_API_KEY
```

2. Build docker:

```
docker-compose build && docker-compose up
```

3. Ingesting SIS Courses into the Local DB
``` bash 
# Attach a shell to the web docker container.
docker exec -it <container_name> bash                       # also possible in vscode

# run cache course script 
# make sure to run in the code/ directory NOT code/data
node data/cacheSISCoursesWithVersions.js
```

View the contents of your local database in MongoDBCompass using the URI `mongodb://localhost:27017/debug`. Don't see it? You may have to [kill off existing processes at port 27017](https://stackoverflow.com/a/18706913). 

### Testing 
The ./tests/routes directory has test files for each of the API route files. Try running `npm test`!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,37 @@
...
+```
+npm i
+``` 
+3. Create .env file at the root. Copy paste .env for development! (Ask a dev for access)
+```
+touch .env
+``` 
+4. Start server.
+```
+npm start
```
Comment (JJamesWWang) [inline README.md:30]: do yall not use nodemon anymore

Comment (mattliu-mygit) [reply README.md:30]: We do; "npm start" runs "nodemon" :0

**Comment by JJamesWWang** [comment]
won't the run locally section fail for new devs because they don't have mongodb installed

**Comment by mattliu-mygit** [comment]
> won't the run locally section fail for new devs because they don't have mongodb installed

Yea, instructions on downloading and running a local mongodb would be super helpful. ^.^ It could be as simple as linking this: https://stackoverflow.com/questions/20796714/how-do-i-start-mongo-db-from-windows

File: README.md [hunk 2]
```diff
@@ -1,4 +1,37 @@
...
+``` 
+5. Navigate to http://localhost:4567/
+
+### Run locally with docker 
 
 To run program from a fresh clone:
```
Comment (mattliu-mygit) [inline README.md:37]: For step 1, do we need to create the .env file again? Maybe modify it to "Update `.env` file with the following:" or "Make sure the `.env` file is created and filled (from earlier step)."

Comment (mattliu-mygit) [reply README.md:37]: Otherwise looks amazing!

**Review by JJamesWWang — APPROVED** [review]
jal boyeo

**Comment by mpark63** [comment]
> jal boyeo

**Comment by mattliu-mygit** [comment]
^what he said

**Comment by mpark63** [comment]
fuck

**Review by mattliu-mygit — COMMENTED** [review]
lgtm

**Comment by mattliu-mygit** [comment]
kewk

**Comment by mpark63** [comment]
oh thank god

**Review by mattliu-mygit — APPROVED** [review]
lmao

**Comment by mpark63** [comment]
not sure what jal boyeo is but tysm!

**Pull request merged**